server/server.js

```diff
@@ -40,13 +40,17 @@ io.on("connection", (socket) => {
 
     console.log(`${userId} joined room ${roomId}`);
 
-    io.sockets.in(roomId).emit("userIsJoined", { users: users });
+    io.sockets.in(roomId).emit("userIsJoined", { users: data, success: true });
```

Comment:
the data you are sending is teh data of a single user not the users in the room.